test/failed-files.test.js

```diff
@@ -16,6 +16,9 @@ const expectForFormat = {
   js(err) {
     expect(err.name).toBe('ReferenceError');
   },
+  'es-module'(err) {
```

Comment:
so `format` in this code is referring to [these](https://github.com/davidtheclark/cosmiconfig/blob/master/src/createExplorer.js#L16).  The purpose of format is to tell cosmiconfig not to [guess and check](https://github.com/davidtheclark/cosmiconfig/blob/master/src/loadDefinedFile.js#L37) the different formats.

You were right not to add `es-module` as a format and so I think this test should be done a different way.  I think a one-off test [underneath this describe block](https://github.com/davidtheclark/cosmiconfig/blob/v5/test/failed-files.test.js#L88) with the same logic you created should do.